node/_fs/_fs_read.ts

```diff
@@ -108,35 +108,26 @@ export function read(
   validatePosition(position);
   validateOffsetLengthRead(offset, length, buffer.byteLength);
 
-  let err: Error | null = null,
-    numberOfBytesRead: number | null = null;
-
-  try {
-    let currentPosition = 0;
-    if (typeof position === "number" && position >= 0) {
-      currentPosition = Deno.seekSync(fd, 0, Deno.SeekMode.Current);
-      Deno.seekSync(fd, position, Deno.SeekMode.Start);
-    }
-
-    numberOfBytesRead = Deno.readSync(fd, buffer);
-
-    if (typeof position === "number" && position >= 0) {
-      Deno.seekSync(fd, currentPosition, Deno.SeekMode.Start);
-    }
-  } catch (error) {
-    err = error instanceof Error ? error : new Error("[non-error thrown]");
-  }
-
   if (!cb) throw new ERR_INVALID_ARG_TYPE("cb", "Callback", cb);
 
-  if (err) {
-    (callback as (err: Error) => void)(err);
-  } else {
-    const data = Buffer.from(buffer.buffer, offset, length);
-    cb(null, numberOfBytesRead ?? 0, data);
-  }
-
-  return;
+  (async () => {
+    try {
+      let nread: number | null;
+      if (typeof position === "number" && position >= 0) {
+        const currentPosition = await Deno.seek(fd, 0, Deno.SeekMode.Current);
+        // The following seek/read/seek calls need to be sync
+        // because those need to happen atomically
```

Comment:
Making these calls synchronous doesn't make any guarantees about atomicity... but the implementation was already using these APIs, so I guess no need to change it here.

Reply:
Right, it was a wrong word. Updated